openpype/lib/avalon_context.py

```diff
@@ -1971,3 +1971,123 @@ def get_last_workfile(
         return os.path.normpath(os.path.join(workdir, filename))
 
     return filename
+
+
+@with_avalon
+def get_linked_ids_for_representations(project, repre_ids, dbcon=None,
+                                       link_type=None, max_depth=0):
+    """Returns list of linked ids of particular type (if provided).
+
+    Goes from representations to version, back to representations
+    Args:
+        project (str)
+        repre_ids (list) or (ObjectId)
+        dbcon (avalon.mongodb.AvalonMongoDB, optional): Avalon Mongo connection
+            with Session.
+        link_type (str): ['reference', '..]
+        max_depth (int): limit how many levels of recursion
+    Returns:
+        (list) of ObjectId - linked representations
+    """
+    if not dbcon:
+        log.debug("Using `avalon.io` for query.")
+        dbcon = avalon.io
+        # Make sure is installed
+        dbcon.install()
+
+    if dbcon.Session["AVALON_PROJECT"] != project:
+        dbcon.Session["AVALON_PROJECT"] = project
+
+    if not isinstance(repre_ids, list):
+        repre_ids = [repre_ids]
+
+    versions = avalon.io.find(
+        {
+            "_id": {"$in": repre_ids},
+            "type": "representation"
+        },
+        projection={"parent": True}
+    )
+    version_ids = [version["parent"] for version in versions]
```

Comment:
Actually, instead of doing that. We could also change the query itself to:
```python
version_id = `io.distinct("parent", {
        "_id": {"$in": repre_ids},
        "type": "representation"
    })
```
Especially if we're just projecting to `parent` only, don't use any other data and only want to know the unique parent ids.


At first I thought it felt wrong storing `version["parent"]` into a variable named `version_ids` is confusing. That doesn't sound like version ids but subset ids. No? But then I noticed the query above. It's not actually querying `versions` but `representations`. So this would actually be `[repre["parent"] for repre in representations]` etc.

Anyway, the `distinct` example solves it either way. :)